crates/matrix-sdk-appservice/src/error.rs

```diff
@@ -17,67 +17,73 @@ use thiserror::Error;
 
 #[derive(Error, Debug)]
 pub enum Error {
```

Comment:
These seem to be a bit misguided I would say. The thing that goes into `#[error()]` should be a human readable message for the error or, if no additional context needs to be provided, the underlying message can be used via `#[error(transparent)]`.

Why is this better?

Reply:
It's better because otherwise the transparent message doesn't contain any context. I'd say the variant is self-describing as for what the context is supposed to be.

Reply:
I'd be also fine with giving the context prefix a more readable name, but I'd say that it wouldn't differ much from the variant name anyway - and this way it's standardized.

Reply:
It should be human readable, like before. The variants without an inner error shouldn't be changed, the other ones could go from

```rust
    #[error(transparent)]
    Http(#[from] http::Error),
```

to

```rust
    #[error("HTTP error: {0}")
    Http(#[from] http::Error),
```

and similar.

Reply:
Like, is there any good reason to not just use the variant as string?

Reply:
I could even imagine upstreaming it to thiserror and introducing something like `transparent_variant_prefix` as macro parameter.

Reply:
Will revert them anyway. More discussion happened here: https://matrix.to/#/!iYnZafYUoXkeVPOSQh:matrix.org/$sBrErzJ4rfkd7ECy6WwEIrvo9rR4LH1ouIIErnn81s8?via=matrix.org&via=famedly.de&via=gnome.org